### PR TITLE
Bug 1780858 - Run a51 browsertime tests on Fenix branch

### PR DESCRIPTION
**Pull request by 92kns**
This patch is to replace G5 with the A51 platform on the Fenix branch

https://bugzilla.mozilla.org/show_bug.cgi?id=1780858

### CODE REVIEW

**Comment by 92kns** [comment]
> No Taskcluster jobs started for this pull request
> 
> ```js
> The `allowPullRequests` configuration for this repository (in `.taskcluster.yml` on the
> default branch) does not allow starting tasks for this pull request.
> ```

How may I get this to run on Taskcluster?

**Comment by Mugurell** [comment]
To run all CI tasks for outside contributors we can push the same changesets to a new PR that would trigger the tasks.
Did that here - https://github.com/mozilla-mobile/fenix/pull/26164.

**Comment by Mugurell** [comment]
I'm seeing some issues related to TC - https://firefox-ci-tc.services.mozilla.com/tasks/JEkNh3GTQsGO4c1uhH3Lcw/runs/0/logs/public/logs/live.log

**Comment by 92kns** [comment]
Thank you @Mugurell for checking!
If I am understanding this log does this mean there is an issue with how the A51 are configured(InsufficientScopes error)? Or is this just a temporary issue with TC?

**Comment by Mugurell** [comment]
Ran the tasks twice and got the same error.
Seems like it has to do with how TC is configured. Maybe @JohanLorenzo can help.

**Comment by JohanLorenzo** [comment]
Hi guys! 👋 

I confirm it's an issue with how Taskcluster is currently configured. By default, we grant workers only to projects who use these workers. This prevents project from taking workers that are not supposed to be theirs. It's especially useful when these workers are in finite numbers and using them elsewhere can result in unwanted delays. Projects are granted access to workers by Taskcluster scopes. That's why we're seeing these errors in the job above:

```
[task 2022-07-25T07:07:45.884Z] * method:     createTask
[task 2022-07-25T07:07:45.884Z] * errorCode:  InsufficientScopes
[task 2022-07-25T07:07:45.884Z] * statusCode: 403
[task 2022-07-25T07:07:45.884Z] * time:       2022-07-25T07:07:45.868Z
[task 2022-07-25T07:07:45.894Z] 2022-07-25 07:07:45,894 - ERROR - Client ID task-client/JEkNh3GTQsGO4c1uhH3Lcw/0/on/us-east-1/i-0b200723f35fcd030/until/1658733905.288 does not have sufficient scopes and is missing the following scopes:
[task 2022-07-25T07:07:45.894Z] 
[task 2022-07-25T07:07:45.894Z] queue:create-task:highest:proj-autophone/gecko-t-bitbar-gw-perf-a51
[task 2022-07-25T07:07:45.894Z] 
```

That said, it's easy to grant these scopes. It's a matter of making a patch like this one[1] in this repo[2]. I'm happy to review it and to help guide you to write it. As far as I can see, it should be a couple of lines to add.


[1] https://phabricator.services.mozilla.com/D26525
[2] https://hg.mozilla.org/ci/ci-configuration/

**Comment by Mugurell** [comment]
Thank you @JohanLorenzo.
Been a while since I last worked with phabricator, opened now https://phabricator.services.mozilla.com/D152742 to grant access to the new a51 worker for all Fenix pull requests. And for other scenarios also since those were using the soon to be deprecated devices.

**Comment by Mugurell** [comment]
@92kns @JohanLorenzo Many new tasks are now running (seems like more than double of what we had until now, do we really want them all running on all PRs?) but they seem to fail with:
> "MOZHARNESS_PATH or MOZHARNESS_URL must be defined"
> "Failed to download and unzip mozharness"

**Comment by 92kns** [comment]
@Mugurell @JohanLorenzo  thanks a lot for your help so far! As for the failures I see that some are failing to time outs. I am working on a patch to increase the process timeout in mozille-central which should hopefully address that https://phabricator.services.mozilla.com/D152890

> Many new tasks are now running (many more than what we had until now, do we really want them all running on all PRs?)

As for this I am not sure what it was like before, however the previous G5's should've been replaced with A51 tests now, so I think it should still be the same amount of tests as before

(And I will of course remove the [on-pull-request] before this is merged )

**Comment by Mugurell** [comment]
These `browsertime` tests weren't running before if we check on other open PRs - eg: https://github.com/mozilla-mobile/fenix/pull/26204

**Comment by 92kns** [comment]
Thanks for clarifying that @Mugurell my mistake, yes what you said is correct not many jobs are running. The intention was originally to see if these tests run and pass before merging with `github-pull-request`. e.g. looking at the commit history of the `kind.yml` this was done before recently https://github.com/mozilla-mobile/fenix/commit/f74909deb1bcc2a020b698c801e5f52f5563183c

and that had many tasks running as well https://github.com/mozilla-mobile/fenix/pull/24984/checks?check_run_id=6272682281

**Comment by 92kns** [comment]
@Mugurell this patch https://phabricator.services.mozilla.com/D152890 has landed in mozilla-central, could I please ask you to retrigger the taskcluster checks?

Afterwards I will push a commit removing `github-pull-request`.

Worth noting there should still be some failures as mentioned in this bug https://bugzilla.mozilla.org/show_bug.cgi?id=1768889#c16 but that is okay and expected

**Comment by Mugurell** [comment]
Not sure why in a previous try to rerun the CI tasks they failed but now everything seems to be running smoothly.

**Comment by 92kns** [comment]
Great, thank you @Mugurell! I see some of the failures are `TypeError: unsupported operand type(s) for *=: 'NoneType' and 'int'` , that should be resolved with a patch landed earlier today https://phabricator.services.mozilla.com/D153089 so that can be ignored

as for the other errors I believe it is the expected ones I was referring to in my previous comment. I will push a commit shortly removing the `github-pull-request` field

**Comment by Mugurell** [comment]
Running all CI tasks. If all is green seems like this can be merged.

File: taskcluster/ci/browsertime/kind.yml [hunk 1]
```diff
@@ -44,10 +44,10 @@ job-defaults:
         platform:
             by-abi:
                 arm64-v8a: android-hw-p2-8-0-android-aarch64-shippable-qr/opt
-                armeabi-v7a: android-hw-g5-7-0-arm7-api-16-shippable-qr/opt
+                armeabi-v7a: android-hw-a51-11-0-arm7-shippable/opt
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:47]: @92kns you'll need to add the `-qr` to match this platform: https://searchfox.org/mozilla-central/source/taskcluster/ci/test/test-platforms.yml#372

See these tests in mozilla-central: https://treeherder.mozilla.org/jobs?repo=mozilla-central&searchStr=android%2Ca51%2Cqr&revision=09bcd3a0096b6d38671c7da6c6980dd9a7e373bb

**Comment by 92kns** [comment]
Hi @Mugurell could I please ask you to retrigger the taskcluster checks? I've had to push a new fix and had to re-add [github-pull-request] to ensure that it works as expected

Thank you!

**Comment by Mugurell** [comment]
> Hi @Mugurell could I please ask you to retrigger the taskcluster checks? I've had to push a new fix and had to re-add `[github-pull-request]` (which I will remove again after the checks) to ensure that it works as expected
> 
> Thank you!

Sure, np, did it now.

> edit: just noticed this option - should I have used `github-push` instead? not that i'd be able to run it since I am an outside contributor, but would it have made things easier for you?

It wouldn't help.
The process for helping outside contributors involves us pushing the same commits - same revisions to a new PR.
CI would run all tasks in our PR because we pushed the changesets and if incidentally the same revisions are in other PRs then they would also see the tasks being run.
In this case I just `pull`ed your recent changes and updated my PR which retriggered all tasks.

**Comment by 92kns** [comment]
Thanks @Mugurell !
I just noticed I don't see taskcluster checks queued up. Do you know why? Does it require the closing/reopening of #26164

**Review by gmierz — APPROVED** [review]
Changes + tests look good now, thanks! You'll need to remove the github-pull-request flag.

**Comment by 92kns** [comment]
Thanks again @Mugurell & @gmierz !
Please merge this

**Pull request merged**